test/e2e/cluster_upgrade_runtimesdk.go

```diff
@@ -176,6 +189,21 @@ func clusterUpgradeWithRuntimeSDKSpec(ctx context.Context, inputGetter func() cl
 			WaitForKubeProxyUpgrade:     input.E2EConfig.GetIntervals(specName, "wait-machine-upgrade"),
 			WaitForDNSUpgrade:           input.E2EConfig.GetIntervals(specName, "wait-machine-upgrade"),
 			WaitForEtcdUpgrade:          input.E2EConfig.GetIntervals(specName, "wait-machine-upgrade"),
+			PreWaitForControlPlaneToBeUpgraded: func() {
```

Comment:
Nit
```suggestion
			BeforeClusterUpgradeWaiter: func() {
```
Same for others below

Reply:
The idea was to frame this more in words which fit to the util functions in test framework vs adding hooks which are RuntimeSDK specific.

Essentially to not couple them too hard to RuntimeSDK so other funcs can use them for other purposes as well